### PR TITLE
Neural net

### PR DESCRIPTION
**Pull request by ayushbihani**

### CODE REVIEW

File: public/css/style.css [hunk 1]
```diff
@@ -687,4 +687,15 @@ canvas{
   stroke-width: 1px;
   stroke: #ccc;
   shape-rendering: crispEdges;
+}
+
+.node:hover {
```
Comment (arjun-rao) [inline public/css/style.css]: What is this for?

File: src/ML/tensorflow.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
+var tf = require("@tensorflow/tfjs");
```
Comment (arjun-rao) [inline src/ML/tensorflow.js]: Rename this file to `neuralnetwork.js`

File: src/ML/tensorflow.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    var loss_ = (loss == undefined)? tfConstants["loss"]:loss;
+    var metrics_ = (metrics == undefined)?tfConstants["metrics"]:metrics;
+    var compile = {
+        optimizer :optimizer_,
+        loss : loss_,
+        metrics : metrics_
+    };
+}
+
+function createModel(){
```
Comment (arjun-rao) [inline src/ML/tensorflow.js]: Rename as NeuralNetwork()

File: src/ML/tensorflow.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+createModel.prototype.compileModel = function(options){
+    this.model.compile(options);
+};
+
+createModel.prototype.predict = function(test){
+    var predict = this.model.predict(tf.tensor([test],[1,[test].length]));
+    return predict;
+};
+
+function train(model,x,y){
```
Comment (arjun-rao) [inline src/ML/tensorflow.js]: Inline this function inside createModel.prototype.train itself

File: src/ML/tensorflow.js [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        }
+        else if (layer["type"] == "compile"){
+            this.model.compile(compileModel(layer["optimizer"],layer["rate"],layer["loss"],layer["optimizer"]));
+        }
+    }
+    console.log(this.model);
+    return this;
+};
+
+createModel.prototype.train = function(y){
```
Comment (arjun-rao) [inline src/ML/tensorflow.js]: Call console.webLog and print the training loss incurred - can we plot it? 

Can we show the loading fidget spinner while it is training?

File: src/ML/tensorflow.js [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    this.y = tf.tensor(y);
+    var model = train(this.model, this.xData, this.y);
+    return model;
+};
+
+createModel.prototype.compileModel = function(options){
```
Comment (arjun-rao) [inline src/ML/tensorflow.js]: Inline this function into createModel() constructor and combine the compile block into the create model block as additional options (like how plot block has options)

File: src/ML/tensorflow.js [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+        metrics : metrics_
+    };
+}
+
+function createModel(){
+    this.model = tf.sequential();
+    this.layerCount = 0;
+}
+
+createModel.prototype.addLayers = function(layers, xData){
```
Comment (arjun-rao) [inline src/ML/tensorflow.js]: Why is data being taken in with adding layers? can't we store the model definition as a JSON and actually create the real tfjs model only in the train block?

File: src/ML/tensorflowjsSamples.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+var tf = require('@tensorflow/tfjs');
```
Comment (arjun-rao) [inline src/ML/tensorflowjsSamples.js]: This seems like a test file - we shouldn't be putting it into version control - or we should put it in a separate folder outside of src/

File: src/milo.js [hunk 9]
```diff
@@ -13,6 +13,8 @@ var MiloStorage = require('./storage');
 var Datasets  = window.Datasets = require('./datasets');
 var Blockly = window.Blockly = require('milo-blocks');
 var Project = require('./project');
+window.nn  = require('./ML/tensorflow');
```
Comment (arjun-rao) [inline src/milo.js]: Why are these in global window scope? Ideally it should only be in the sandbox.

File: views/toolbox.html [hunk 10]
```diff
@@ -400,6 +400,7 @@
       <category name="Vector Math" colour="%{BKY_COLOUR_HUE}">
         <block type="tf_number"></block>
         <block type="tf_tensor"></block>
+        <block type="test"></block>
```
Comment (arjun-rao) [inline views/toolbox.html]: This block name needs to be more descriptive
